src/main/java/dev/dkvl/womutils/panel/WomPanel.java

```diff
@@ -436,4 +364,89 @@ private void openPlayerProfile(String username)
 
         SwingUtilities.invokeLater(() -> LinkBrowser.browse(url));
     }
+
+    private JPanel createButtonsPanel()
+    {
+        JPanel buttonsPanel = new JPanel();
+        buttonsPanel.setLayout(new GridBagLayout());
+        buttonsPanel.setBorder(new EmptyBorder(0, 0, 10, 0));
+
+        GridBagConstraints gbc = new GridBagConstraints();
+        gbc.fill = GridBagConstraints.HORIZONTAL;
+        gbc.gridx = 0;
+        gbc.weightx = 1;
+        gbc.ipady = 10;
+
+        JButton updateButton = new JButton();
+        updateButton.setFont(FontManager.getRunescapeSmallFont());
+        updateButton.setEnabled(false);
+        updateButton.addActionListener(e ->
+            womClient.updatePlayer(sanitize(searchBar.getText())));
+        updateButton.setText("Update");
+
+        JButton profileButton = new JButton();
+        profileButton.setFont(FontManager.getRunescapeSmallFont());
+        profileButton.setEnabled(false);
+        profileButton.addActionListener(e ->
+            openPlayerProfile(sanitize(searchBar.getText())));
+        profileButton.setText("Open Profile");
+
+        buttons.add(updateButton);
+        buttons.add(profileButton);
+
+        buttonsPanel.add(updateButton, gbc);
+        gbc.gridx++;
+        gbc.insets.left = 7;
+        buttonsPanel.add(profileButton, gbc);
+
+        return buttonsPanel;
+    }
+
+    private JPanel createOverViewPanel()
+    {
+        JPanel miscInfoPanel = new JPanel();
+        miscInfoPanel.setLayout(new GridLayout(3, 2, 5, 5));
+
+        for (MiscInfo infoLabel : miscInfoLabels)
+        {
+            JLabel label = infoLabel.getLabel();
+            ImageIcon icon;
+
+            switch (infoLabel)
+            {
+                case COUNTRY:
```

Comment:
The icon can be a field of the enum entries and once again be directly used by the wrapper object.